src/mon_markings.c

```diff
@@ -0,0 +1,377 @@
+        sMenu->tileLoadState++;
+        break;
+    case 13:
+        CpuFastCopy(sMenu->frameTiles + TILE_SIZE_4BPP * 6, dest, TILE_SIZE_4BPP);
+        for (i = 0; i < 6; i++)
+        {
+            CpuFastCopy(sMenu->frameTiles + TILE_SIZE_4BPP * 7, dest + TILE_SIZE_4BPP * (i + 1), TILE_SIZE_4BPP);
+        }
+        CpuFastCopy(sMenu->frameTiles + TILE_SIZE_4BPP * 8, dest + TILE_SIZE_4BPP * 7, TILE_SIZE_4BPP);
+        sMenu->tileLoadState++;
+        return FALSE;
+    case 14:
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+void sub_80F7404(void)
+{
+    sub_80F728C();
+    while (sub_80F72D4())
+        ;
+}
+
+void sub_80F7418(u8 markings, s16 x, s16 y)
+{
+    u16 i;
+    sMenu->cursorPos = 0;
+    sMenu->markings = markings;
+    for (i = 0; i < 4; i++)
+        sMenu->markingsArray[i] = (sMenu->markings >> i) & 1;
+    DestroyMenuCursor();
+    sub_80F761C(x, y, sMenu->baseTileTag, sMenu->basePaletteTag);
+}
+
+void sub_80F7470(void)
+{
+    u16 i;
+
+    for (i = 0; i < 3; i++)
+    {
+        FreeSpriteTilesByTag(sMenu->baseTileTag + i);
+    }
+
+    FreeSpritePaletteByTag(sMenu->basePaletteTag);
+    FreeSpritePaletteByTag(sMenu->basePaletteTag + 1);
+
+    for (i = 0; i < 2; i++)
+    {
+        if (!sMenu->menuWindowSprites[i])
+            return;
+        DestroySprite(sMenu->menuWindowSprites[i]);
+    }
+
+    for (i = 0; i < 4; i++)
+    {
+        if (!sMenu->menuMarkingSprites[i])
+            return;
+        DestroySprite(sMenu->menuMarkingSprites[i]);
+    }
+
+    DestroyMenuCursor();
+
+    if (sMenu->menuTextSprite)
+        DestroySprite(sMenu->menuTextSprite);
+}
+
+bool8 sub_80F7500(void)
+{
+    u16 i;
+    struct SpriteSheet sheets[3] =
+    {
+        { sMenu->menuWindowSpriteTiles, 0x1000, sMenu->baseTileTag },
+        { gPokenavConditionMenuMisc_Gfx, 0x520, sMenu->baseTileTag + 1 },
+        { NULL, 0, 0 }
+    };
+
+    if (sMenu->spriteSheetLoadRequired)
+    {
+        LoadTilesForSpriteSheets(sheets);
+        sMenu->spriteSheetLoadRequired = FALSE;
+    }
+
+    if (gMain.newKeys & DPAD_UP)
+    {
+        s8 pos;
+        PlaySE(SE_SELECT);
+        pos = --sMenu->cursorPos;
+        if (pos < 0)
+            sMenu->cursorPos = 5;
+        return TRUE;
+    }
+
+    if (gMain.newKeys & DPAD_DOWN)
+    {
+        s8 pos;
+        PlaySE(SE_SELECT);
+        pos = ++sMenu->cursorPos;
+        if (pos > 5)
+            sMenu->cursorPos = 0;
+        return TRUE;
+    }
+
+    if (gMain.newKeys & A_BUTTON)
+    {
+        PlaySE(SE_SELECT);
+
+        switch (sMenu->cursorPos)
+        {
+        case 4:
+            sMenu->markings = 0;
+            for (i = 0; i < 4; i++)
+                sMenu->markings |= sMenu->markingsArray[i] << i;
+            return FALSE;
+        case 5:
+            return FALSE;
+        }
+
+        sMenu->markingsArray[sMenu->cursorPos] = sMenu->markingsArray[sMenu->cursorPos] ? FALSE : TRUE;
```

Comment:
`sMenu->markingsArray[sMenu->cursorPos] = !sMenu->markingsArray[sMenu->cursorPos];` works here, if you find that more readable.